lightning/lightning.go

```diff
@@ -143,78 +216,279 @@ func (l *Lightning) Stop() {
 	l.shutdown()
 }
 
+func writeJSONError(w http.ResponseWriter, code int, prefix string, err error) {
+	type errorResponse struct {
+		Error string `json:"error"`
+	}
+
+	w.WriteHeader(code)
+
+	if err != nil {
+		prefix += ": " + err.Error()
+	}
+	json.NewEncoder(w).Encode(errorResponse{Error: prefix})
+}
+
+func parseTaskID(req *http.Request) (int64, string, error) {
+	path := strings.TrimPrefix(req.URL.Path, "/")
+	taskIDString := path
+	verb := ""
+	if i := strings.IndexByte(path, '/'); i >= 0 {
+		taskIDString = path[:i]
+		verb = path[i+1:]
+	}
+
+	taskID, err := strconv.ParseInt(taskIDString, 10, 64)
+	if err != nil {
+		return 0, "", err
+	}
+
+	return taskID, verb, nil
+}
+
 func (l *Lightning) handleTask(w http.ResponseWriter, req *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
+	if l.taskCfgs == nil {
+		writeJSONError(w, http.StatusNotImplemented, "server-mode not enabled", nil)
```

Comment:
is the response message right?

Reply:
It's right, `l.taskCfgs` is not nil only if `--server-mode` is enabled. Added comments to explain.